src/cloud.c

```diff
@@ -276,15 +276,15 @@ int cloud_register_device(const char *id, const char *name)
 	const char *json_str;
 	int result;
 
-	jobj_device = parser_device_json_create(id, name);
-	json_str = json_object_to_json_string(jobj_device);
-
 	queue_cloud = amqp_declare_new_queue(AMQP_QUEUE_CLOUD);
 	if (!queue_cloud.bytes) {
 		hal_log_error("Error on declare a new queue.\n");
 		return -1;
 	}
 
+	jobj_device = parser_device_json_create(id, name);
+	json_str = json_object_to_json_string(jobj_device);
```

Comment:
Please create another patch verifying if `jobj_device` is NULL

Reply:
Ok. Will be in another PR.

Reply:
Ack